lib/src/main/java/graphql/nadel/engine/plan/NadelExecutionPlanFactory.kt

```diff
@@ -33,25 +41,30 @@ internal class NadelExecutionPlanFactory(
     ): NadelExecutionPlan {
         val executionSteps = mutableListOf<AnyNadelExecutionPlanStep>()
 
-        traverseQuery(rootField) { field ->
-            transforms.forEach { transform ->
-                val state = transform.isApplicable(
-                    executionContext,
-                    executionBlueprint,
-                    services,
-                    service,
-                    field,
-                    serviceHydrationDetails,
-                )
-                if (state != null) {
-                    executionSteps.add(
-                        NadelExecutionPlan.Step(
+        executionContext.timer.batch { timer ->
+            traverseQuery(rootField) { field ->
+                transformsWithTimingStepInfo.forEach { (transform, timingStep) ->
+                    val state = timer.time(step = timingStep) {
```

Comment:
Added timing for execution plan and will add some more later

Reply:
At some point a lambda in a lambda in a lambda makes for header to understand code

```
executionContext.timer.batch { timer ->
            traverseQuery(rootField) { field ->
                transformsWithTimingStepInfo.forEach { (transform, timingStep) ->
                    val state = timer.time(step = timingStep) {
                    
```

I think its good to assigned the "target code - into a function holder and then batch time that

````
var transformCode = traverseQuery(rootField) { field -> ...
```

That said the closures over timer and timeStep here make that a little tricky

But as a general rule - if you have 3 layers of labdas call each other - consider an assignment OR a method that returns the  lambda say